### ISSUE TITLE
The label of the training sample is displayed incorrectly

### ISSUE THREAD

**Issue by Sea814** [ultralytics/ultralytics#21390]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I recently trained a custom dataset on yoloworld, this dataset has a large number of categories, more than 1000, and I deleted some training samples of category IDs (e.g. 0, 2, etc.), but in the train_batch0.jpg pictures generated during training, I found that the category IDs of bbox are messy. I have made sure that my training data is correct, pls what is the reason for this?

![Image](https://github.com/user-attachments/assets/39c4d1e3-74cb-4894-b58c-26df583403de)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Sea814, thank you for bringing this to our attention and for your interest in Ultralytics 🚀!

Please note this is an automated response to help get you started—an Ultralytics engineer will follow up with you soon to assist further.

If this is a 🐛 Bug Report, could you please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate? This will make it much easier for our team to debug the issue. For your case, please include:
- A snippet of your dataset YAML
- A few example label files and images
- The exact command you used to start training
- Any modifications made to the dataset or code

If you’re working with a custom dataset, please ensure you follow the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) and provide as much detail as possible, including training logs and sample data.

Check out our [Docs](https://docs.ultralytics.com/) for more info, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage.

Join our community for real-time help or discussion:
- [Discord](https://discord.com/invite/ultralytics) 🎧
- [Discourse](https://community.ultralytics.com/)
- [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

Please ensure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in any of these verified environments (all dependencies, including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Duplicate of https://github.com/ultralytics/ultralytics/issues/18142

**Comment by Sea814** [comment]
> Duplicate of [#18142](https://github.com/ultralytics/ultralytics/issues/18142)

I get it, thanks to your reply

**Comment by glenn-jocher** [comment]
You're welcome! Yes, this is the same issue as #18142 where class IDs can appear to change due to augmentations. The training itself should work correctly, but the visualization may show confusing labels.
